### PR TITLE
Auth feature

### PR DESCRIPTION
**Pull request by ifonya105**
fixed auth with the token
added conversation list test
added is_im property
added user property to channel
fixed upload file method
fixed UserUIInteraction test

### CODE REVIEW

**Comment by mckunda** [comment]
@ifonya105 `Channel` [already has `is_im` and `user` fields](https://github.com/Inumedia/SlackAPI/blob/master/SlackAPI/Channel.cs), and that's why the appveyor build has failed.

**Comment by ifonya105** [comment]
> @ifonya105 `Channel` [already has `is_im` and `user` fields](https://github.com/Inumedia/SlackAPI/blob/master/SlackAPI/Channel.cs), and that's why the appveyor build has failed.

yeah, it's strange that I didn't notice it.
I'll fix

**Comment by we4sz** [comment]
When can we expect this? It's blocking my current development :)

**Comment by ifonya105** [comment]
Someone please accept the request.

**Comment by AdamCLarsen** [comment]
> When can we expect this? It's blocking my current development :)

This is blocking my project as well, would be helpful to get this out.

File: SlackAPI/SlackClientBase.cs [hunk 1]
```diff
@@ -111,9 +130,18 @@ protected HttpWebRequest CreateWebRequest(Uri requestUri)
             return httpWebRequest;
         }
 
-        protected HttpResponseMessage PostRequest(string requestUri, MultipartFormDataContent form)
+        protected Task<HttpResponseMessage> PostRequestAsync(string requestUri, MultipartFormDataContent form, string token)
```
Comment (Inumedia) [inline SlackAPI/SlackClientBase.cs:133]: FYI, this is the part I take issue with. We have a split client for handling async workflows specifically.

Calling `.Result` does not work as a way to invoke async from a sync call

Comment (Inumedia) [reply SlackAPI/SlackClientBase.cs:133]: And I'm realizing we did do `.Result` before. Ugh.

**Review by Inumedia — APPROVED** [review]

**Pull request merged**